### PR TITLE
fix: now using @aws/dynamodb-data-mapper for queries

### PR DESCRIPTION
**Pull request by MacMcIrish**

### CODE REVIEW

File: src/modules/model.js [hunk 1]
```diff
@@ -36,90 +40,94 @@ module.exports = ({
...
-      throw EntryNotFound({ id });
+    let resp;
+    try {
+      resp = await mapper.get(Object.assign(new Model(), { id }), {
+        projection: objectFields.split(fields),
+        readConsistency: 'strong'
+      });
+    } catch (err) {
+      if (err.name === 'ItemNotFoundException') {
+        throw EntryNotFound({ id });
```
Comment (simlu) [inline src/modules/model.js]:
```suggestion
        throw ItemNotFound({ id });
```

File: src/modules/model.js [hunk 2]
```diff
@@ -1,25 +1,29 @@
...
   awsConfig = {},
   errorMap: {
-    EntryNotFound = DefaultEntryNotFound,
-    EntryExists = DefaultEntryExists
+    ItemNotFound = DefaultItemNotFound,
+    ItemExists = DefaultItemExists
   } = {},
   callback = () => {}
 }) => {
+  class Model {}
```
Comment (simlu) [inline src/modules/model.js]: honestly looks like `id` should be part of the model constructor

File: src/modules/model.js [hunk 3]
```diff
@@ -36,90 +40,94 @@ module.exports = ({
...
-        ExpressionAttributeValues: { ':id': { S: id } },
-        TableName: tableName,
-        Item: dynamodbConverter.marshall(Object.assign({}, data, { id }))
-      }, {
-        expectedErrorCodes: ['ConditionalCheckFailedException']
-      });
-      if (resp === 'ConditionalCheckFailedException') {
-        throw EntryExists({ id });
+      try {
+        await mapper.put(Object.assign(new Model(), data, { id }), {
```
Comment (simlu) [inline src/modules/model.js]: We might want to wrap `Model` in a closure and have dedicated functions to update etc. I think that will make it a lot cleaner and future proof

File: src/modules/model.js (hunk 2 above)
Comment (MacMcIrish) [reply src/modules/model.js]: They internally instantiate the model when unmarshalling the data, I believe that's why they have a basic class in their docs. I personally believe we should stick to that pattern, but we can also wrap this in another class?
```
class Model {
  constructor(id) {
    class DataMapperClass {}
    this.DataMapperClass = DataMapperClass;
    ...
```

File: src/modules/model.js [hunk 4]
```diff
@@ -1,125 +1,154 @@
...
+    errorMap: {
+      ItemNotFound = DefaultItemNotFound,
+      ItemExists = DefaultItemExists
+    } = {},
+    callback = () => {
+    }
+  }) {
+    class InternalDataMapperClass {
+    }
+
```
Comment (simlu) [inline src/modules/model.js]:
```suggestion
```

File: src/modules/model.js [hunk 5]
```diff
@@ -1,125 +1,154 @@
...
+    awsConfig,
+    errorMap: {
+      ItemNotFound = DefaultItemNotFound,
+      ItemExists = DefaultItemExists
+    } = {},
+    callback = () => {
+    }
+  }) {
+    class InternalDataMapperClass {
+    }
```
Comment (simlu) [inline src/modules/model.js]:
```suggestion
```

File: src/modules/model.js [hunk 6]
```diff
@@ -1,125 +1,154 @@
...
+    schema,
+    awsConfig,
+    errorMap: {
+      ItemNotFound = DefaultItemNotFound,
+      ItemExists = DefaultItemExists
+    } = {},
+    callback = () => {
+    }
+  }) {
+    class InternalDataMapperClass {
```
Comment (simlu) [inline src/modules/model.js]:
```suggestion
    class InternalDataMapperClass {}
```

File: src/modules/model.js [hunk 7]
```diff
@@ -1,125 +1,154 @@
...
-  const precheck = () => {
-    Object.entries({ modelName, tableName })
+    schema,
+    awsConfig,
+    errorMap: {
+      ItemNotFound = DefaultItemNotFound,
+      ItemExists = DefaultItemExists
+    } = {},
+    callback = () => {
+    }
```
Comment (simlu) [inline src/modules/model.js]:
```suggestion
```

File: src/modules/model.js [hunk 8]
```diff
@@ -1,125 +1,154 @@
...
-  });
-  const precheck = () => {
-    Object.entries({ modelName, tableName })
+    schema,
+    awsConfig,
+    errorMap: {
+      ItemNotFound = DefaultItemNotFound,
+      ItemExists = DefaultItemExists
+    } = {},
+    callback = () => {
```
Comment (simlu) [inline src/modules/model.js]:
```suggestion
    callback = () => {}
```

File: src/modules/model.js [hunk 9]
```diff
@@ -1,125 +1,154 @@
...
+    this.mapper = mapper;
+    this.modelName = modelName;
+    this.tableName = tableName;
+    this.InternalDataMapperClass = InternalDataMapperClass;
+    this.ItemNotFound = ItemNotFound;
+    this.ItemExists = ItemExists;
+    this.callback = callback;
+  }
+
+  callbackWrapper({ id, actionType }) {
```
Comment (simlu) [inline src/modules/model.js]: I don't think these should be named params? Just `cb('get', id)`?

**Pull request merged**